lib/Conversion/ExportVerilog/PrepareForEmission.cpp

```diff
@@ -510,18 +511,84 @@ static bool reuseExistingInOut(Operation *op) {
   return true;
 }
 
+/// Return true if this is something that will get printed as a unary operator
+/// by the Verilog printer.
+/// TODO: This is copied from PrettifyVerilog. Create a common library for
+/// verilog expression stuffs.
+static bool isVerilogUnaryOperator(Operation *op) {
+  if (isa<comb::ParityOp>(op))
+    return true;
+
+  if (auto xorOp = dyn_cast<comb::XorOp>(op))
+    return xorOp.isBinaryNot();
+
+  if (auto icmpOp = dyn_cast<comb::ICmpOp>(op))
+    return icmpOp.isEqualAllOnes() || icmpOp.isNotEqualZero();
+
+  return false;
+}
+
+/// Return true if it is beneficial to spill the operation under the specified
+/// spilling heuristic.
+bool shouldSpillWire(Operation &op,
```

Comment:
What I anticipate we will eventually want is something that associates a cost/badness to a line and then breaks it if the computed value exceeds the badness amount which is configurable based on specific user-defined penalties.

It would be good to go look at clang-format and get some idea of its options and how it uses them in its line splitting (which is admittedly different from wire spilling).

Reply:
> What I anticipate we will eventually want is something that associates a cost/badness to a line and then breaks it if the computed value exceeds the badness amount which is configurable based on specific user-defined penalties. 

Yes, I agree that would be right direction to go. I've implemented size estimation mechanism for namehint heurstic, that would be able to be extended to cost/badness based framework in the future.